src/pybind/mgr/dashboard/frontend/src/app/shared/services/notification.service.ts

```diff
@@ -0,0 +1,84 @@
+import { Injectable } from '@angular/core';
+
+import * as _ from 'lodash';
+import { ToastsManager } from 'ng2-toastr';
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+
+import { NotificationType } from '../enum/notification-type.enum';
+import { CdNotification } from '../models/cd-notification';
+
+@Injectable()
+export class NotificationService {
+  // Observable sources
+  private notificationDataSource = new BehaviorSubject<CdNotification[]>([]);
+
+  // Observable streams
+  notificationData$ = this.notificationDataSource.asObservable();
+
+  KEY = 'cdNotifications';
+
+  constructor(public toastr: ToastsManager) {
+    const stringNotifications = localStorage.getItem(this.KEY);
```

Comment:
How about `notifications: string[]=`?

Reply:
``stringNotifications`` is the stringify version of notifications, that is saved on localStorage.
Don't think its the best approach to change its name to ``notifications``.

On the other hand I have changed ``parseNotifications`` to ``notifications`` because in that case they really are notifications.